content/en/continuous_delivery/deployments/argocd.md

```diff
@@ -0,0 +1,141 @@
+data:
+  template.cd-visibility-template: |
+    webhook:
+      cd-visibility-webhook:
+        method: POST
+        body: |
+            {
+              "app": {{toJson .app}},
+              "context": {{toJson .context}},
+              "service_type": {{toJson .serviceType}},
+              "recipient": {{toJson .recipient}},
+              "commit_metadata": {{toJson (call .repo.GetCommitMetadata .app.status.operationState.syncResult.revision)}}
+            }
+```
+
+<div class="alert alert-warning">
+The call to populate the <code>commit_metadata</code> field is not required, and it's used to enrich the payload with Git information.
+In case you are using Helm repositories as the source of your Argo CD application, adjust the body by removing that line.
+</div>
+
+`cd-visibility-template` is the name of the template, and `cd-visibility-webhook` is a reference to the service created above.
+
+The third step is to create the trigger, again in the same `argocd-notifications-cm` config map:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-notifications-cm
+data:
+  trigger.cd-visibility-trigger: |
+    - when: app.status.operationState.phase in ['Succeeded', 'Failed', 'Error', 'Running'] and app.status.health.status in ['Healthy', 'Degraded']
+      send: [cd-visibility-template]
+```
+
+`cd-visibility-trigger` is the name of the trigger, and `cd-visibility-template` is a reference to the template created above.
+
+After the service, trigger and template have been created, you can subscribe any of your Argo CD application to the integration.
+This is done via modifying the annotations of the application, either via the Argo CD UI or via modifying the application definition:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  annotations:
+    notifications.argoproj.io/subscribe.cd-visibility-trigger.cd-visibility-webhook: ""
+    dd_env: <YOUR_ENV>
+```
+
+There are two annotations:
+1. The notifications annotation subscribes the application to the notification setup created above.
+2. The `dd_env` annotation configures the environment of the application. Replace `YOUR_ENV` above with the environment
+   to which this application is deploying (for example: `staging`, `prod`). If you don't set this annotation,
+   the environment will be defaulted to `none`.
+
+Once this final step is completed, the integration is done, and you can monitor your Argo CD deployments in Datadog.
+
+## Visualize deployments in Datadog
+
+The [Deployments][6] and [Deployment Executions][7] pages populate with data after a deployment is executed.
+
+## Troubleshooting
+
+In case the notifications are not sent, a good starting point are the logs of the `argocd-notification-controller` pod.
+The controller logs when it's sending a notification (`Sending notification ...`) and when it fails to notify a recipient
+(`Failed to notify recipient ...`). For additional troubleshooting scenarios, see the [Argo CD docs][8].
```

Comment:
```suggestion
In case the notifications are not sent, examine the logs of the `argocd-notification-controller` pod. The controller logs when it is sending a notification (for example: `Sending notification ...`) and when it fails to notify a recipient
(for example: `Failed to notify recipient ...`). For additional troubleshooting scenarios, see the [official Argo CD documentation][8].
```